### PR TITLE
CI Update

### LINKED ISSUE #2: Add standard Rust workflows
The CI for this repository currently runs linting, formatting, and cargo-deny. It should also

- [x] run cargo build
- [x] run tests and doctests
- [x] build documentation (with warnings)

We enable [some extra warnings](https://github.com/boltlabs-inc/key-mgmt/blob/develop/lock-keeper/src/lib.rs#L7-L11) to catch common programming errors (missing debug, unused results and types, soon-to-be-deprecated types, and broken doc links). These should be added.

We might also want to update the formatting to match our overall repo formatting, which enables some nightly lints. See [key-mgmt rustfmt.toml](https://github.com/boltlabs-inc/key-mgmt/blob/develop/rustfmt.toml). 

For references on how to run these, see [key-mgmt rust CI](https://github.com/boltlabs-inc/key-mgmt/blob/develop/.github/workflows/rust.yml).

### PR DESCRIPTION
**Pull request by tyurek**
This PR updates the CI to run `cargo doc` and add the extra warnings from issue #2. It additionally updates the codebase to pass the new checks.

This should close #2

### CODE REVIEW

**Comment by marsella** [comment]
Per offline conversation, can you update this to use `cargo make` like `key-mgmt` currently does?

**Comment by tyurek** [comment]
So I went ahead and made a makefile which uses similar flags to key-mgmt. Looks like there's at least one more flag that I wasn't previously handling, so I'll do some more code fixing

**Comment by tyurek** [comment]
Okay, this should be ready for review now. I've changed the base for readability purposes and it looks like the last CI test I ran is available

File: src/lib.rs [hunk 1]
```diff
@@ -23,19 +23,22 @@
...
+//! process a message from another participant, producing a set of outgoing
+//! messages that in turn must be delivered to other participants.
 //!
 //! For an example of how to actually integrate this library into a higher-level
 //! application that handles the communication between participants in parallel,
 //! take a look at the provided [network example](./examples/network/README.md).
-//!
-//!
 
 #![allow(non_snake_case)] // FIXME: To be removed in the future
```
Comment (gatoWololo) [inline src/lib.rs:35]: lol

**Review by gatoWololo — APPROVED** [review]
Looks good to me!

**Pull request merged**